src/tests/test_utils_bitvector.cpp

```diff
@@ -0,0 +1,1053 @@
+/*
+ * (C) 2023-2024 Jack Lloyd
+ * (C) 2023-2024 René Meusel, Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include "tests.h"
+
+#include <botan/internal/bitvector.h>
+#include <botan/internal/fmt.h>
+
+#include <algorithm>
+
+namespace Botan_Tests {
+
+namespace {
+
+std::vector<Test::Result> test_bitvector_bitwise_accessors() {
+   return {
+      Botan_Tests::CHECK("default constructed bitvector",
+                         [](auto& result) {
+                            Botan::bitvector bv;
+                            result.confirm("default constructed bitvector is empty", bv.empty());
+                            result.test_eq("default constructed bitvector has zero size", bv.size(), size_t(0));
+                         }),
+
+      Botan_Tests::CHECK("preallocated construction of bitvector",
+                         [](auto& result) {
+                            Botan::bitvector bv(10);
+                            result.confirm("allocated bitvector is not empty", !bv.empty());
+                            result.test_eq("allocated bitvector has allocated size", bv.size(), size_t(10));
+                            for(size_t i = 0; i < 10; ++i) {
+                               result.confirm("bit not set yet", !bv.at(i));
+                            }
+                         }),
+
+      Botan_Tests::CHECK("setting bits",
+                         [](auto& result) {
+                            Botan::bitvector bv(32);
+                            bv.set(1);
+                            bv.at(2) = true;
+                            bv.set(3);
+
+                            result.confirm("0", !bv.at(0));
+                            result.confirm("1", bv.at(1));
+                            result.confirm("2", bv.at(2));
+                            result.confirm("3", bv.at(3));
+                            result.confirm("4", !bv.at(4));
```

Comment:
Good idea. I applied randomized tests for `set`, `unset`, and `flip` since these are the lowest-hanging fruits for this. Also, I think these are among the tests most prone to edge-case oversights.

Reply:
Thanks